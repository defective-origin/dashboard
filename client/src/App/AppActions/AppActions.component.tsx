import React from 'react'

// ---| core |---
import { cn } from 'tools'

// ---| pages |---
// ---| screens |---
// ---| components |---
import Aside from 'components/Aside'
import Portal from 'components/Portal'

// ---| self |---
import css from './AppActions.module.scss'

export type AppActionsProps = {
  className?: string
  children?: React.ReactNode
}

/**
 * Component description.
 *
 * How to use
 * @example
 * <AppActions />
 */
export function AppActions(props: AppActionsProps): JSX.Element {
  const { children, className, ...otherProps } = props
  const _className = cn(css.AppActions, className)

  return (
    <Aside className={_className} area='right' justifies='end' {...otherProps}>
      <Portal.Container name='page-actions' />

      {children}
    </Aside>
  )
}

AppActions.displayName = 'AppActions'

export default AppActions
