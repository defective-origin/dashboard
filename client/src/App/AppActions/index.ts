/* INJECT_IMPORT_PLACE */

/* INJECT_DEFAULT_EXPORT_PLACE */
export { default } from './AppActions.component'

/* INJECT_EXPORT_PLACE */
export * from './AppActions.component'
