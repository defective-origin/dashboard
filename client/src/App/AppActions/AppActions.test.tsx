// ---| tests |---
import { render } from 'tests'

// ---| self |---
import AppActions from './AppActions.component'
// import useAppActions from './AppActions.hook'
// import AppActionsProvider, AppActions from './AppActions.context'

describe('[AppActions] component', () => {
  it('should render component', () => {
    const container = render(<AppActions />)

    expect(container.snapshot()).toMatchSnapshot()
  })
})

// describe('[AppActions] hook', () => {
//   it('should return options', () => {
//     const { result } = renderHook(() => useAppActions())
//
//     expect(result.current).toEqual(null)
//   })
// })

// describe('[AppActions] context', () => {
//   const wrapper = (props: AppActionsProps) => <AppActionsProvider { ...props } />
// 
//   it('should return options', () => {
//     const { result } = renderHook(() => useAppActions(), { wrapper })
//
//     expect(result.current).toBeTruthy()
//   })
// })
