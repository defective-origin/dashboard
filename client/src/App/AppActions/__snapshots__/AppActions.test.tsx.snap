// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[AppActions] component > should render component 1`] = `
<DocumentFragment>
  <div
    class="block block--y item _Aside_5b7d5a _AppActions_664bb3"
    style="justify-content: end; grid-area: right;"
  >
    <div
      class="_PortalContainer_8d45f7"
      id="portal-page-actions"
    />
  </div>
</DocumentFragment>
`;
